libraries/AP_InertialSensor/AP_InertialSensor_LSM9DS1.h

```diff
@@ -0,0 +1,102 @@
+#ifndef __AP_INERTIAL_SENSOR_LSM9DS1_H__
+#define __AP_INERTIAL_SENSOR_LSM9DS1_H__
+
+#include <cstdio>
+
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/SPIDevice.h>
+
+#include "AP_InertialSensor.h"
+#include "AP_InertialSensor_Backend.h"
+
+
+/* enable debug to see a register dump on startup */
+#define LSM9DS1_DEBUG 0
+
+
+
+class AP_InertialSensor_LSM9DS1 : public AP_InertialSensor_Backend
+{
+
+public:
+    virtual ~AP_InertialSensor_LSM9DS1() { }
+    void start(void) override;
+    bool update() override;
+
+    static AP_InertialSensor_Backend *probe(AP_InertialSensor &imu,
+                                            AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev,
+                                            enum Rotation rotation = ROTATION_NONE);
+
+private:
+    AP_InertialSensor_LSM9DS1(AP_InertialSensor &imu,
+                              AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev,
+                              int drdy_pin_num_xg,
+                              enum Rotation rotation);
+
+    static AP_InertialSensor_Backend *detect(AP_InertialSensor &imu);
+
+    struct PACKED sensor_raw_data {
```

Comment:
Is packing necessary here? The alignment should be identical with 3 ints?

Reply:
Packed is not necessary here, but left for consistency. I can get rid of it if need be.